api/src/controllers/policies.ts

```diff
@@ -0,0 +1,220 @@
+import { ErrorCode, isDirectusError } from '@directus/errors';
+import type { PrimaryKey } from '@directus/types';
+import express from 'express';
+import { respond } from '../middleware/respond.js';
+import useCollection from '../middleware/use-collection.js';
+import { validateBatch } from '../middleware/validate-batch.js';
+import { MetaService } from '../services/meta.js';
+import { PoliciesService } from '../services/policies.js';
+import asyncHandler from '../utils/async-handler.js';
+import { sanitizeQuery } from '../utils/sanitize-query.js';
+
+const router = express.Router();
+
+router.use(useCollection('directus_policies'));
+
+router.post(
+	'/',
+	asyncHandler(async (req, res, next) => {
+		const service = new PoliciesService({
+			accountability: req.accountability,
+			schema: req.schema,
+		});
+
+		const savedKeys: PrimaryKey[] = [];
+
+		if (Array.isArray(req.body)) {
+			const keys = await service.createMany(req.body);
+			savedKeys.push(...keys);
+		} else {
+			const key = await service.createOne(req.body);
+			savedKeys.push(key);
+		}
+
+		try {
+			if (Array.isArray(req.body)) {
+				const items = await service.readMany(savedKeys, req.sanitizedQuery);
+				res.locals['payload'] = { data: items };
+			} else {
+				const item = await service.readOne(savedKeys[0]!, req.sanitizedQuery);
+				res.locals['payload'] = { data: item };
+			}
+		} catch (error: any) {
+			if (isDirectusError(error, ErrorCode.Forbidden)) {
+				return next();
+			}
+
+			throw error;
+		}
+
+		return next();
+	}),
+	respond,
+);
+
+const readHandler = asyncHandler(async (req, res, next) => {
+	const service = new PoliciesService({
+		accountability: req.accountability,
+		schema: req.schema,
+	});
+
+	const metaService = new MetaService({
+		accountability: req.accountability,
+		schema: req.schema,
+	});
+
+	let result;
+
+	// TODO fix this at the service level
+	const temporaryQuery = { ...req.sanitizedQuery, limit: -1 };
+
+	if (req.body.keys) {
+		result = await service.readMany(req.body.keys, temporaryQuery);
+	} else {
+		result = await service.readByQuery(temporaryQuery);
+	}
+
+	const meta = await metaService.getMetaForQuery('directus_policies', temporaryQuery);
+
+	res.locals['payload'] = { data: result, meta };
+	return next();
+});
+
+router.get('/', validateBatch('read'), readHandler, respond);
+router.search('/', validateBatch('read'), readHandler, respond);
+
+router.get(
+	'/me/flags',
```

Comment:
Anyone have a better name for this? Since in the future it might not only be flags. Should it follow the naming of the libs and maybe be called `/me/global`? The idea behind this endpoint is to accumulate any properties that are derived when taking all policies of a user into account, like global access and currently the `enfore_tfa` flag.

Reply:
Mhh agreed `/me/globals` sounds better than flags to me too. 

> The idea behind this endpoint is to accumulate any properties

Mhh so maybe `/me/properties` ? :smile: 

_If_ we dont expand on it `/me/access` sounds pretty fitting for right now, but yeah that might be come less good in the future.

Reply:
Hmm, I personally dislike just `/me` because the endpoint does not return any policies, but some aggregation of all policies of the user, and we might end up with an actual `/me` endpoint that returns all policies at some point? wdyt?

Reply:
Can you give an quick example of how the current response looks like?

Reply:
It is effectively a condensed version of whatever flags we have on policies right now, that apply to a user.

```json
{ "app_access": true, "admin_access": false, "enforce_tfa": false }
```

Reply:
Okay, I think `global(s)` sounds more appropriate, that's ultimately what we're retrieving here: global flags/properties/whatever.

>  but some aggregation of all policies of the user, and we might end up with an actual `/me` endpoint that returns all policies at some point? wdyt?

You mean at a later point or is that something you'd like to look into now?

Reply:
At a later point. Currently I don't see the need of retrieving all the policies of a user efficiently, as the app does not need to know them to work. But maybe that need arises at some point in the future, so I'd like to keep that option alive.